### DISCUSSION TITLE
Inject header with every request

### DISCUSSION THREAD

**Discussion by X7md** [vercel/swr#2527] [Q&A] [answered]
Is there any way to do some kind of `middleware` to inject authorization header with every request?

The problem is I use state management lib `zustand` to deal with JWT header, so I can't inject it outside react `rules of hooks`, aka I can't make reusable fetcher.

**Comment by koba04** [comment]
Yes, you can do it with `middleware` like https://swr.vercel.app/docs/middleware#request-logger , or you can also pass down the JWT headers as React Context or pass a fetcher having JWT headers into SWRConfig and use it in your component.

**Comment by mnik01** [comment, marked as answer]
You can use zustand store data outside the react component. They have this API

useStore().getState(s => s.someField)
